lib/Controller/ChattyLLMController.php

```diff
@@ -0,0 +1,362 @@
+	}
+
+	/**
+	 * @param string $content
+	 * @param int $timestamp
+	 * @param ?string $title
+	 * @return JSONResponse
+	 */
+	#[NoAdminRequired]
+	public function newSession(string $content, int $timestamp, ?string $title = null): JSONResponse {
+		if ($this->userId === null) {
+			return new JSONResponse(['error' => $this->l10n->t('User not logged in')], Http::STATUS_UNAUTHORIZED);
+		}
+
+		$user = $this->userManager->get($this->userId);
+		if ($user === null) {
+			return new JSONResponse(['error' => $this->l10n->t('User not found')], Http::STATUS_UNAUTHORIZED);
+		}
+
+		$userInstructions = $this->config->getAppValue(
+			Application::APP_ID,
+			'chat_user_instructions',
+			$this->l10n->t(Application::CHAT_USER_INSTRUCTIONS),
+		);
+		$userInstructions = str_replace('{user}', $user->getDisplayName(), $userInstructions);
+
+		try {
+			$session = new Session();
+			$session->setUserId($this->userId);
+			$session->setTitle($title);
+			$session->setTimestamp($timestamp);
+			$this->sessionMapper->insert($session);
+
+			$systemMsg = new Message();
+			$systemMsg->setSessionId($session->getId());
+			$systemMsg->setRole($this->l10n->t('system'));
+			$systemMsg->setContent($userInstructions);
+			$systemMsg->setTimestamp($session->getTimestamp());
+			$this->messageMapper->insert($systemMsg);
+
+			$humanMsg = new Message();
+			$humanMsg->setSessionId($session->getId());
+			$humanMsg->setRole($this->l10n->t('human'));
+			$humanMsg->setContent($content);
+			$humanMsg->setTimestamp($session->getTimestamp());
+			$this->messageMapper->insert($humanMsg);
+
+			return new JSONResponse($session);
+		} catch (\OCP\DB\Exception | \RuntimeException $e) {
+			$this->logger->warning('Failed to create a chat session', ['exception' => $e]);
+			return new JSONResponse(['error' => $this->l10n->t('Failed to create a chat session')], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	/**
+	 * @param integer $sessionId
+	 * @param string $title
+	 * @return JSONResponse
+	 */
+	#[NoAdminRequired]
+	public function updateSession(int $sessionId, string $title): JSONResponse {
+		if ($this->userId === null) {
+			return new JSONResponse(['error' => $this->l10n->t('User not logged in')], Http::STATUS_UNAUTHORIZED);
+		}
+
+		try {
+			$this->sessionMapper->updateSession($sessionId, $title);
+			return new JSONResponse();
+		} catch (\OCP\DB\Exception | \RuntimeException  $e) {
+			$this->logger->warning('Failed to update the chat session', ['exception' => $e]);
+			return new JSONResponse(['error' => $this->l10n->t('Failed to update the chat session')], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	/**
+	 * @param integer $sessionId
+	 * @return JSONResponse
+	 */
+	#[NoAdminRequired]
+	public function deleteSession(int $sessionId): JSONResponse {
+		if ($this->userId === null) {
+			return new JSONResponse(['error' => $this->l10n->t('User not logged in')], Http::STATUS_UNAUTHORIZED);
+		}
+
+		try {
+			$this->sessionMapper->deleteSession($sessionId);
```

Comment:
QueryMapper exposes a delete method already, so you could just 
```suggestion
			$session = $this->sessionMapper->find($sessionId);
			$this->sessionMapper->delete($session);
```

Reply:
Cool! I was missing the find method in my thought process.

Reply:
looks like a query has to be built for find anyway so this way should be fine ig?